integration/spark-common/src/main/scala/org/apache/spark/rdd/CarbonMergeFilesRDD.scala

```diff
@@ -145,7 +169,7 @@ class CarbonMergeFilesRDD(
       if (isHivePartitionedTable) {
         CarbonLoaderUtil
           .mergeIndexFilesInPartitionedSegment(carbonTable, split.segmentId,
-            segmentFileNameToSegmentIdMap.get(split.segmentId))
+            segmentFileNameToSegmentIdMap.get(split.segmentId), split.partitionPath)
```

Comment:
Each partition based task may write & overwrite the same segment file with only its partition path.

Reply:
In case of partition table, for a segment we will be writing segment files for each of the partition as merge index is per partition of a segment. Finally, we will be merging all the segment files to a single segment file and the same will be updated to the table status. Same is used here. Executors will write the segment file for the partition it is handling, at last driver merge all these segment files of partition to a segment.